src/librbd/io/ImageRequestWQ.h

```diff
@@ -22,6 +22,31 @@ class AioCompletion;
 template <typename> class ImageRequest;
 class ReadResult;
 
+class ImageQosStatus {
+public:
+  ImageQosStatus() : throttling_active(false) {}
+
+  utime_t get_prev_limit() const {
+    return prev_limit;
+  }
+
+  void set_prev_limit(utime_t limit) {
+    prev_limit = limit;
+  }
+
+  void set_throttling_flag(bool flag) {
+    throttling_active = flag;
+  }
+
+  bool get_throttling_flag() {
+    return throttling_active;
+  }
+
+private:
+  utime_t prev_limit;
```

Comment:
Nit: is there a better name for this variable (and getters/setters)? Perhaps something like ```last_io_time```?